Mod Configuration Menu/Mods/BG3MCM/ScriptExtender/Lua/Client/IMGUILayer.lua

```diff
@@ -35,101 +45,156 @@ function IMGUILayer:SetClientStateValue(settingId, value, modGUID)
-            Ext.Mod.GetMod(modGUID).Info.Name ..
-            "' for visibility condition. Please contact " ..
-            Ext.Mod.GetMod(modGUID).Info.Author .. " about this issue.")
-        return false
+
+function IMGUILayer:UpdateVisibilityOfUiElement(modGUID, uiElement)
+    local visibilityTriggers = self.visibilityTriggers[modGUID] or {}
+    local modSettings = self.mods[modGUID].settingsValues
+    local visibleIf = visibilityTriggers[uiElement]
+    local logicalOperator = visibleIf.LogicalOperator or "and"
+    local visible = true
+    if logicalOperator == "or" then
+        visible = false
     end
 
-    local strValue, strTrigger = tostring(value), tostring(triggerValue)
-    local numValue, numTrigger = tonumber(value), tonumber(triggerValue)
+    for _, condition in ipairs(visibleIf.Conditions) do
+        local settingIdTriggering = condition.SettingId
+        local operator = condition.Operator
+        local triggerValue = condition.ExpectedValue
 
-    local operators = {
-        ["=="] = function(a, b) return a == b end,
-        ["!="] = function(a, b) return a ~= b end,
-        ["<="] = function(a, b) return a <= b end,
-        [">="] = function(a, b) return a >= b end,
-        ["<"] = function(a, b) return a < b end,
-        [">"] = function(a, b) return a > b end
-    }
+        local value = nil
+        if operator == "isVisible" then
+            value = self.uiElementByName[modGUID][settingIdTriggering].Visible
+        else
+            value = modSettings[settingIdTriggering]
+        end
 
-    if operators[operator] then
-        if operator == "==" or operator == "!=" then
-            return operators[operator](strValue, strTrigger)
+        local strValue, strTrigger = tostring(value), tostring(triggerValue)
+        local numValue, numTrigger = tonumber(value), tonumber(triggerValue)
+
+        local v = nil
+        if operator == "==" or operator == "!="  or operator == "isVisible" then
+            v = self.operators[operator](strValue, strTrigger)
         elseif numValue ~= nil and numTrigger ~= nil then
-            return operators[operator](numValue, numTrigger)
+            v = self.operators[operator](numValue, numTrigger)
+        else
+            MCMWarn(0,
+                    "Something go wrong, contact MCM support.") -- should never happen (trigegr is not empty at that point, value from widget too)
+            v = false
         end
-        return false
-    end
 
-    MCMWarn(0, "Unknown comparison operator: " .. operator .. " for mod '" ..
-        Ext.Mod.GetMod(modGUID).Info.Name ..
-        "'. Please contact " ..
-        Ext.Mod.GetMod(modGUID).Info.Author .. " about this issue.")
-    return false
+        if v and (logicalOperator == "or") then
+            visible = true
+            break
+        elseif (not v) and (logicalOperator == "and") then
+            visible = false
+            break
+        end
```

Comment:
Why is this being evaluated in an if statement?

Reply:
because of the break to stop as soon as possible. It's not a simple assignment